### ISSUE TITLE
Spell Check does not work like GG1

### ISSUE THREAD

**Issue by okrick** [DistributedProofreaders/guiguts-py#343]
1. GG2 spell check examines the entire document, even when only a portion is selected, unlike GG1 which checks only the selected text.

2. Unlike GG1's popup, GG2 presents a full list of misspelled words in a new window, similar to the Word Frequency -> Check Spelling function.

3. GG2 lists every occurrence of a misspelled word, for instance, showing all 1647 appearances of "png" separately. This differs from GG1, which displays each word once with its total count. The extensive GG2 list (20,352 entries) is overwhelming compared to GG1's slightly more manageable list of 10,215 unique words.

4. If emulating Word Frequency -> Spelling Check, GG2 should allow sorting by word frequency and length. Longer words often have more spelling errors. Additionally, ignoring frequently misspelled words can be helpful, assuming they are intentional style choices.

5. The GG1 Spell Check "Add Goodwords to Proj. Dictionary" functionality should be replicated in GG2.

**Comment by windymilla** [comment]
The first thing to note is that GG2's spellcheck is based on GG1's Spell Query. The old GG1 Spell check uses Aspell, which is problematic on Windows as it doesn't support utf8 properly. I don't intend to use Aspell in GG2.

1. Although the GG1 Spell Query doesn't restrict its check to the selected range (as Spell Check does in GG1, though I didn't know about that feature), I don't see any reason why GG2's spell checker can't be limited to the selected range.
2. That is like GG1 Spell Query.
3. That is not accurate, I don't think. GG1 Spell Check also handles each occurrence of a word, not just one for every "png". It's clear it has to do that because some occurrences of "png" might be correct, and some incorrect, so the Spell Check has to be able to handle that. I suspect what happens is that when you see it is a word like "png" where you know all occurrences are OK, you do Skip All, or something similar in GG1 Spell Check. That's OK - you can do the same in GG2 Spelling Checker - just Shift Right Click a "png" warning message and they will all be removed from the list. Or you can click Skip All, or Add to Project Dictionary - both of those will remove all the warnings about "png" with one click.
4. It's not emulating WF Spelling Check. However GG1 Spell Query has a feature that GG1 Spell Check does not, that many people have found helpful, and from your comment, you would too. It's "threshold" - you can say "if an apparent bad spelling occurs 3 or more times, then assume it's good". That feature hasn't yet been implemented in GG2 Spelling checker, but it will be.
5. "Add Goodwords to Proj. Dictionary" is already in GG2, but it's in the File-->Project menu, because it can be useful for things other than Spell Check.

In summary, if you agree with my comments above, I think the following still need to be done:
a. Restrict GG2 spell checking to the selected range (or whole file if nothing selected).
b. Add "threshold" to the spell check dialog.

**Comment by windymilla** [comment]
Thanks for the additional info - it may be that with some additions, the Spell Query style will be acceptable too. If not, we'll have to review the way the GG2 spell checking works.
